addOns/client/src/main/java/org/zaproxy/addon/client/internal/ClientMapWriter.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2025 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.client.internal;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.zaproxy.addon.client.ExtensionClientIntegration;
+import org.zaproxy.addon.client.ui.PopupMenuExportClientMap;
+import org.zaproxy.zap.utils.Stats;
+
+public final class ClientMapWriter {
+
+    private static final Logger LOGGER = LogManager.getLogger(ClientMapWriter.class);
+
+    private static final String NODE_KEY = "node";
+    private static final String ROOT_NODE_NAME = "ClientMap";
+    private static final String CHILDREN_KEY = "children";
+    private static final String STORAGE_KEY = "isStorage";
```

Comment:
I'd suggest `storage` for consistency with `visited`. Why these have constants?

Reply:
⭐